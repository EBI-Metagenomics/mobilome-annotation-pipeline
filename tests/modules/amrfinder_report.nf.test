nextflow_process {

    name "Test Process AMRFINDER_REPORT"
    script "modules/amrfinder_report.nf"
    process "AMRFINDER_REPORT"

    // FIXME: this is testing an empty file

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/tests/test_data/mobilome_results/func_annot/amrfinderplus.tsv")
                input[1] = file("${projectDir}/tests/test_data/mobilome_results/mobilome_prokka.gff")
                input[2] = file("${projectDir}/tests/test_data/mobilome_results/preprocessing/contigID.map")
                input[3] = file("${projectDir}/tests/test_data/integrator_inputs/no_user_gff")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
