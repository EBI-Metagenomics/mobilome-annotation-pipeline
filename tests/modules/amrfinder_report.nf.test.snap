{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "amr_location.tsv:md5,8e266a07c592d7f5d4373efa9b62e648"
                ]
            }
        ],
        "timestamp": "2024-05-29T12:48:41.869549688"
    }
}